fe1-web/model/objects/__tests__/Chirp.test.ts

```diff
@@ -0,0 +1,96 @@
+import 'jest-extended';
+import '__tests__/utils/matchers';
+import {
+  Chirp, ChirpState,
+} from '../Chirp';
+import { Timestamp } from '../Timestamp';
+import { Hash } from '../Hash';
+
+const TIMESTAMP = new Timestamp(12345);
+const HASH_ID = new Hash('id');
+const HASH_PARENT = new Hash('parent');
+
+describe('Chirp object', () => {
+  it('can do a state round trip', () => {
+    const chirpState: ChirpState = {
+      id: '1234',
+      sender: 'me',
+      text: 'text',
+      time: 1234,
+      likes: 0,
+      parentId: '5678',
+    };
+    const chirp = Chirp.fromState(chirpState);
+    expect(chirp.toState()).toStrictEqual(chirpState);
+  });
+
+  it('throws an error when id is undefined', () => {
+    const createWrongChirp = () => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const wrongChirp = new Chirp({
+        sender: 'me',
+        text: 'text',
+        time: TIMESTAMP,
+        likes: 0,
+        parentId: HASH_PARENT,
+      });
+    };
+    expect(createWrongChirp).toThrow(Error);
+  });
+
+  it('throws an error when sender is undefined', () => {
+    const createWrongChirp = () => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const wrongChirp = new Chirp({
+        id: HASH_ID,
+        text: 'text',
+        time: TIMESTAMP,
+        likes: 0,
+        parentId: HASH_PARENT,
+      });
+    };
+    expect(createWrongChirp).toThrow(Error);
+  });
+
+  it('throws an error when text is undefined', () => {
+    const createWrongChirp = () => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const wrongChirp = new Chirp({
```

Comment:
I believe you can avoid this pattern like so:

```suggestion
    const createWrongChirp = () => new Chirp({
```

The arrow functions would then return a chirp, instead of declaring an unused variable.

Reply:
That's exactly what I was searching for, thanks a lot !